L1Trigger/L1TZDC/plugins/L1TZDCProducer.cc

```diff
@@ -0,0 +1,4885 @@
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/Stage2Layer2FirmwareFactory.h"
+#include "L1Trigger/L1TCalorimeter/interface/Stage2MainProcessor.h"
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/CaloParamsHelper.h"
+#include "CondFormats/DataRecord/interface/L1TCaloParamsRcd.h"
+#include "CondFormats/DataRecord/interface/L1TCaloParamsO2ORcd.h"
+
+#include "DataFormats/L1TCalorimeter/interface/CaloTower.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+#include "DataFormats/L1Trigger/interface/Tau.h"
+#include "DataFormats/L1Trigger/interface/Jet.h"
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+
+//
+// class declaration
+//
+
+using namespace l1t;
+
+class L1TZDCProducer : public edm::stream::EDProducer<> {
+public:
+  explicit L1TZDCProducer(const edm::ParameterSet& ps);
+  ~L1TZDCProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  void endRun(edm::Run const&, edm::EventSetup const&) override;
+
+  int zdcLUTIndexHelper(int iDetPos, int iBXPos);
+
+  //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+  // ----------member data ---------------------------
+
+  // input tokens
+  //Add the ZDC token - remove other tokens later when more certain what is not needed i.e. at direction of L1Trigger maintainers, particularly the Stage2Layer2Producer maintainers
+  edm::EDGetTokenT<QIE10DigiCollection> m_zdcToken;
+  edm::ESGetToken<CaloParams, L1TCaloParamsRcd> m_candidateToken;
+
+  // put tokens
+  edm::EDPutTokenT<EtSumBxCollection> m_etToken;
+
+  //Following the L1TStage2Layer2Producer
+  CaloParamsHelper* m_params;
+  int m_scaleFactor;
+
+  bool m_doHardCodeLUT;
+};
+
+L1TZDCProducer::L1TZDCProducer(const edm::ParameterSet& ps) {
+  // register what you produce
+  m_etToken = produces<EtSumBxCollection>("zdcEtSums");
+
+  // register what you consume and keep token for later access:
+  m_zdcToken = consumes<QIE10DigiCollection>(ps.getParameter<edm::InputTag>("zdcToken"));
+  m_candidateToken = esConsumes<CaloParams, L1TCaloParamsRcd, edm::Transition::BeginRun>();
+
+  // placeholder for the parameters
+  m_params = new CaloParamsHelper;
+
+  m_doHardCodeLUT = ps.getParameter<bool>("doHardCodeLUT");
+}
+
+L1TZDCProducer::~L1TZDCProducer() { delete m_params; }
```

Comment:
Once `m_params` is either a smart pointer or used by value, this can be removed, and the declaration above can be set to `= default;`

Reply:
Done in latest commit.